absl/time/internal/cctz/src/time_zone_impl.cc

```diff
@@ -70,14 +70,13 @@ bool time_zone::Impl::LoadTimeZone(const std::string& name, time_zone* tz) {
     }
   }
 
-  // Load the new time zone (outside the lock).
-  std::unique_ptr<const Impl> new_impl(new Impl(name));
-
   // Add the new time zone to the map.
   std::lock_guard<std::mutex> lock(TimeZoneMutex());
   if (time_zone_map == nullptr) time_zone_map = new TimeZoneImplByName;
   const Impl*& impl = (*time_zone_map)[name];
   if (impl == nullptr) {  // this thread won any load race
+    // Load the new time zone (outside the lock).
+    std::unique_ptr<const Impl> new_impl(new Impl(name));
```

Comment:
This change is a performance regression. There is no reason to allocate this inside the lock. You've even moved the comment that says "do this outside the lock" into the locked section.

Was this a manual change, or was it generated by an automated tool?

Reply:
@derekmauro,
I might not have noticed my corrections, but in this block I did it in vain

Reply:
@derekmauro,
fix it https://github.com/abseil/abseil-cpp/commit/926c63e38d30956bbb74c3d34ee54aedb8173c4e